ktor-server/ktor-server-cio/jvmAndNix/src/io/ktor/server/cio/CIOApplicationEngine.kt

```diff
@@ -35,78 +37,25 @@ public class CIOApplicationEngine(
         public var connectionIdleTimeoutSeconds: Int = 45
     }
 
-    private val configuration = Configuration().apply(configure)
-
-    private val corePoolSize: Int = maxOf(
-        configuration.connectionGroupSize + configuration.workerGroupSize,
-        environment.connectors.size + 1 // number of selectors + 1
-    )
+    private val configuration: Configuration by shared(Configuration().apply(configure))
 
     @OptIn(InternalCoroutinesApi::class)
-    private val engineDispatcher = ExperimentalCoroutineDispatcher(corePoolSize)
+    private val engineDispatcher = Dispatchers.IOBridge
```

Comment:
Is it expected, that now base parameters of `Configuration` will not affect CIO dispatchers on JVM anymore?

Reply:
Yep, `ExperimentalCoroutineDispatcher` should be dropped now and the dispatcher should be controlled using `kotlinx.coroutines` system property

Reply:
Or, may be after coroutines 1.6 we can use `limitedParallelism`?

https://github.com/Kotlin/kotlinx.coroutines/blob/develop/kotlinx-coroutines-core/common/src/CoroutineDispatcher.kt#L97-L101

Reply:
I will keep `corePoolSize` and implement this as soon as kotlinx.coroutines 1.6 is released